mystery_word.py

```diff
@@ -0,0 +1,206 @@
+# Jason Aylward
+# Homework 3
+import random
+import nightmare
+
+
+default_wordlist = "/usr/share/dict/words"
+mystery_word = ""
+discovered_letters = ""
+guessed_letters = ""
+scoreboard = ""
+guesses = 8
+difficulty = 0
+
+
+def start_game():
+    """Prints the the Introduction and
+    Queries the user for the difficulty level"""
+    print("Welcome to Mystery Word!\nEnter 'Quit' to Quit anytime.")
+    print("""Choose your level of difficulty
+            (E)asy
+            (M)edium
+            (H)ard
+            (N)ightmare""")
+    difficulty = choose_difficulty(input(">>").lower())
+    return difficulty
+
+
+def choose_difficulty(mode):
+    """Allows user to enter difficulty and runs recursively on bad input"""
+    if mode == "e" or mode == "easy":
+        return "e"
+    if mode == "m" or mode == "medium":
+        return "m"
+    if mode == "h" or mode == "hard":
+        return "h"
+    if mode == "n" or mode == "nightmare":
+        return "n"
+    if mode == "q" or mode == "quit":
+        return "q"
+    else:
+        print("Try Again\n")
+        return choose_difficulty(input(">>").lower())
+
+
+def get_wordlist(file_address):
+    """opens file and reads it as one large string
+    builds a list of words by splitting the string on ' '. """
+    with open(file_address) as file:
+        file_string = file.read()
+    word_list = file_string.split()
+    return word_list
+
+
+def generate_word(file_address):
+    """receives the word list from get_wordlist()
+    chooses random word within the constraints of word_length
+    and is_common_noun()
+    If user has chosen to quit or move to nightmare mode, return 0"""
+    if difficulty == "q" or difficulty == "n":
+        return 0
+    wordlist = get_wordlist(file_address)
+    random_int = random.randint(0, len(wordlist)-1)
+    random_word = wordlist[random_int]
+    if is_common_noun(random_word) and is_proper_word_length(random_word):
+        print("Your mystery word "
+              "is {} characters long.".format(len(random_word)))
+        return random_word
+    else:
+        return generate_word(file_address)
+
+
+def is_common_noun(word):
+    """ Checks if word is a common noun.
+    Prevents redundancies such as Titanic and titanic. """
+    if word == word.lower():
+        return True
+    else:
+        return False
```

Comment:
Whenever you have a statement like:

```py
if predicate:
  return True
else:
  return False
```

or its opposite, you can simplify it down to:

```py
return predicate
```

For this, you could have:

```py
return word == word.lower()
```